File: src/main/java/io/qross/psql/ForToLoop.java
package io.qross.psql;

public class ForToLoop {

    public String variable; //自变量
    public String rangeBegin;
    public String rangeEnd;

    public Statement statement;

    public ForToLoop(Statement statement, String variable, String rangeBegin, String rangeEnd) {
        this.statement = statement;
        this.variable = Statement.removeVariableModifier(variable);
        this.rangeBegin = rangeBegin;
        this.rangeEnd = rangeEnd;
    }

    public int parseBegin() {
        return Integer.parseInt(this.statement.parseStandardSentence(this.rangeBegin));
    }

    private int parseEnd() {
        return Integer.parseInt(this.statement.parseStandardSentence(this.rangeEnd));
    }

    public boolean hasNext() {
        return Integer.parseInt((String)statement.PSQL.findVariableValue(this.variable).value) <= this.parseEnd();
    }
}

File: src/main/scala/io/qross/psql/ForToLoop.scala
package io.qross.psql

import io.qross.util.TypeExt._

class ForToLoop(statement: Statement, var variable: String, rangeBegin: String, rangeEnd: String) {
    variable = variable.removeVariableModifier()

    def parseBegin: Integer = {
        statement.parseStandardSentence(rangeBegin).toInt
    }

    def parseEnd: Integer = {
        statement.parseStandardSentence(rangeEnd).toInt
    }

    def hasNext: Boolean = {
        statement.PSQL.findVariableValue(variable).value.asInstanceOf[String].toInt <= parseEnd
    }
}
